examples/src/test/java/io/numaproj/numaflow/examples/sink/forkjoin/ConcurrentSinkTest.java

```diff
@@ -0,0 +1,293 @@
+            testIterator.addDatum(createTestDatum("id-3", "valid-message-3"));
+            testIterator.addDatum(createTestDatum("id-4", "valid-message-4"));
+            
+            ResponseList responseList = concurrentSink.processMessages(testIterator);
+            
+            assertNotNull(responseList);
+            assertEquals(4, responseList.getResponses().size());
+            
+            // Verify that we get responses for all messages
+            for (Response response : responseList.getResponses()) {
+                assertNotNull(response.getId());
+                assertTrue(response.getId().startsWith("id-"));
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("Performance Tests")
+    class PerformanceTests {
+
+        @Test
+        @DisplayName("Should demonstrate concurrent processing benefits")
+        void testConcurrentProcessingPerformance() {
+            int messageCount = 50;
+            
+            // Prepare test data
+            SinkerTestKit.TestListIterator sequentialIterator = new SinkerTestKit.TestListIterator();
+            SinkerTestKit.TestListIterator concurrentIterator = new SinkerTestKit.TestListIterator();
+            
+            for (int i = 0; i < messageCount; i++) {
+                String message = "performance-test-message-" + i;
+                sequentialIterator.addDatum(createTestDatum("seq-id-" + i, message));
+                concurrentIterator.addDatum(createTestDatum("con-id-" + i, message));
+            }
+            
+            // Test with single thread (effectively sequential)
+            ConcurrentSink sequentialSink = new ConcurrentSink(1);
+            long sequentialStart = System.currentTimeMillis();
+            ResponseList sequentialResult = sequentialSink.processMessages(sequentialIterator);
+            long sequentialTime = System.currentTimeMillis() - sequentialStart;
+            sequentialSink.shutdown();
+
+            // Test with multiple threads (concurrent processing)
+            ConcurrentSink concurrentSink = new ConcurrentSink(4);
+            long concurrentStart = System.currentTimeMillis();
+            ResponseList concurrentResult = concurrentSink.processMessages(concurrentIterator);
+            long concurrentTime = System.currentTimeMillis() - concurrentStart;
+            concurrentSink.shutdown();
+            
+            // Verify both produce same number of results
+            assertEquals(messageCount, sequentialResult.getResponses().size());
+            assertEquals(messageCount, concurrentResult.getResponses().size());
+            
+            System.out.printf("Sequential processing: %d ms%n", sequentialTime);
+            System.out.printf("Concurrent processing: %d ms%n", concurrentTime);
+            
+            // Note: In unit tests, the overhead might make concurrent processing slower
+            // for small datasets, but this test demonstrates the framework works
```

Comment:
Since the test case is named `Should demonstrate concurrent processing benefits` - can we use a large dataset and verify concurrentTime is shorter than sequentialTime?